drivers: nxp_s32_canxl: add CANXL MRU handler

congnguyenhuu opened the pull request:
Add CANXL MRU handler, use the same IRQ number as RX, TX interrupt. Update the error priority that is lower priority than the the tx_rx_mru priority incase the error interrupt happens continuously, mru interrupt priority must be higher to get report error counter. Otherwise the mru interrupt can be delayed by error interrupt and never call to MRU handler. This fixes #75022.

henrikbrixandersen commented:
> Add CANXL MRU handler, use the same RX, TX IRQ number. Update the error priority that is lower priority than the the tx_rx_mru priority incase the error interrupt happens continuously, mru interrupt priority must be higher to get report error counter. Otherwise the mru interrupt can be delayed by error interrupt and never call to MRU handler. This fixes #75022.

This is a bug fix. 

> Additionally, enable CANXL_INT_RXSMB_OVER and CAN_INT_ERR_FAST(ERR for CAN-FD). Statistic CAN errors and count CAN TX and RX errors via BCANXL that uses CANXL MRU to exchange informations with CAN host.

This is a feature addition. 

Please split this in at least two commits, preferably two PRs. We only accept bug fixes for v3.7.0 currently, so a PR combining a bug fix with a feature addition cannot go in.

henrikbrixandersen reviewed (changes requested):
Please split this as described above.

congnguyenhuu committed:
drivers: nxp_s32_canxl: add CANXL MRU handler

Add CANXL MRU handler, use the same RX, TX IRQ number.
Update the error priority that is lower priority than
the the tx_rx_mru priority incase the error interrupt
happens continuously, mru interrupt priority must be
higher to get report error counter. Otherwise the mru
interrupt can be delayed by error interrupt and
never call to MRU handler. This fixes #75022.

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

congnguyenhuu commented:
> Please split this as described above.

I separated this to 2 PRs. this PR only includes change for bug fix now

manuargue commented on drivers/can/can_nxp_s32_canxl.c:
```diff
@@ -1003,11 +1004,16 @@ static int can_nxp_s32_init(const struct device *dev)
 
 	Canexcel_Ip_RxTxIRQHandler(config->instance);
+
+	if ((config->base_mru->CHXCONFIG[0u].CH_CFG0 & CANXL_MRU_CH_CFG0_CHE_MASK)
```
Where is this MRU channel enabled?

manuargue replied:
Nevermind, I see you already detailed this in the issue description.

manuargue reviewed (approved):
No objections from my side
